src/wp-includes/class-wp-query.php

```diff
@@ -2542,6 +2542,8 @@ public function get_posts() {
 				$post_type_where = " AND {$wpdb->posts}.post_type IN ('" . implode( "', '", array_map( 'esc_sql', $in_search_post_types ) ) . "')";
 			}
 		} elseif ( ! empty( $post_type ) && is_array( $post_type ) ) {
+			// Sort post types to ensure same cache key generation.
+			sort( $post_type );
```

Comment:
I assume this needs to happen here as well as sorting the `post_type` args in `generate_cache_key()` in order to normalize the SQL as well?

Reply:
>I assume this needs to happen here as well as sorting the post_type args in generate_cache_key() in order to normalize the SQL as well?

Correct. A query for ```post_type IN ( 'page', 'post' )``` would get the same results as ```post_type IN ( 'post', 'page' )```

Reply:
It's outside the diff by in the `any` block above this one:

```
$in_search_post_types = get_post_types( array( 'exclude_from_search' => false ) );
sort( $in_search_post_types );
```

As with this change, it will normalize the SQL